### PR TITLE
added tryouts action

### PR DESCRIPTION
**Pull request by alicertel**
Added new action to upload build to [Tryouts.io](https://tryouts.io/)

``` ruby
tryouts(
  api_token: "...",
  app_id: "application-id",
  build_file: "test.ipa"
)
```

Thanks,
Ali

### CODE REVIEW

**Comment by alicertel** [comment]
I don't think  tests fail, because of the changes i made. I did test this in my local environment with `rake test` with ruby 2.0.0, it passes all the tests. Any Thoughts?

**Comment by jcampbell05** [comment]
Don't see any reason for it breaking, tried a `bundle install` I've had issues like that when I haven't updated the bundle.lock and the version of the gem has changed. 

Any ideas @KrauseFx ?

File: docs/Actions.md [hunk 1]
```diff
@@ -764,6 +764,22 @@ appetize(
...
+tryouts(
+  api_token: "...",
+  app_identifier: "application-hash",
+  build_file: "test.ipa",
+)
+```
+
+Additionally you can specify `status`, `notify` and `notes_path` or `notes`.
+
+For more information about the available options, check out `fastlane action tryouts` or head over to [Tryouts Documentation](http://tryouts.readthedocs.org/en/latest/releases.html#create-release) for our complete API docs.
```
Comment (KrauseFx) [inline docs/Actions.md]: Please change to 
```
For more information about the available options, run `fastlane action tryouts` or check out the [Tryouts Documentation](http://tryouts.readthedocs.org/en/latest/releases.html#create-release)
```

File: docs/Actions.md [hunk 2]
```diff
@@ -764,6 +764,22 @@ appetize(
...
+
+```ruby
+tryouts(
+  api_token: "...",
+  app_identifier: "application-hash",
+  build_file: "test.ipa",
+)
+```
+
+Additionally you can specify `status`, `notify` and `notes_path` or `notes`.
```
Comment (KrauseFx) [inline docs/Actions.md]: You can remove this line

File: lib/fastlane/actions/tryouts.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        end
+      end
+
+      def self.description
+        "Upload a new build to Tryouts"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :app_identifier,
```
Comment (KrauseFx) [inline lib/fastlane/actions/tryouts.rb]: `app_identifier` is e.g. `tools.fastlane.app`, I assume you're talking about a hash here? If so, please change to `app_id` or similar

File: lib/fastlane/actions/tryouts.rb [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                                       raise "Couldn't find notes file at path '#{value}'".red unless File.exist?(value)
+                                     end,
+                                     optional: true),
+          FastlaneCore::ConfigItem.new(key: :notify,
+                                     env_name: "TRYOUTS_NOTIFY",
+                                     description: "Notify testers? 1 for yes",
+                                     default_value: "1"),
+          FastlaneCore::ConfigItem.new(key: :status,
+                                     env_name: "TRYOUTS_STATUS",
+                                     description: "2 to make your release public. Release will be distributed to available testers. 1 to make your release private. Release won't be distributed to testers. This also prevents release from showing up for SDK update",
```
Comment (KrauseFx) [inline lib/fastlane/actions/tryouts.rb]: You might want to add a `verify_block` for this to make sure no other number is passed

File: lib/fastlane/actions/tryouts.rb [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        options = {}
+        options[:build] = Faraday::UploadIO.new(params[:build_file], 'application/octet-stream')
+
+        if params[:notes_path]
+          options[:notes] = File.read(params[:notes_path])
+        else
+          options[:notes] = params[:notes] if params[:notes]
+        end
+
+        options[:notify] = params[:notify]
```
Comment (KrauseFx) [inline lib/fastlane/actions/tryouts.rb]: Do you need a `to_s` here? Or does your back-end supports number and strings here?

Comment (KrauseFx) [inline lib/fastlane/actions/tryouts.rb]: The user might just use the action using `tryouts(notify: 1)` (passing an integer)

**Comment by KrauseFx** [comment]
Don't worry about the tests, they were broken last week (my fault, sorry). 

Could you resolve the merge conflict (probably Actions.md)? Also I added some comments before it's ready to be merged.

Thanks for your contribution @alicertel :+1:

**Comment by alicertel** [comment]
Thank you so much for the reviews @KrauseFx . I resolved the merge conflict and updated the action as you suggested.

**Comment by KrauseFx** [comment]
This looks great, thanks for working on the new action @alicertel :tada:

**Pull request merged**